fix web3 connection sync routine if nethermind:

jordipainan opened the pull request:
Adapt the sync progress routine to work properly
when the connected web3 endpoint runs the Nethermind
codebase. The eth_syncing JSON RPC call returns 0x0
as block number.

Add an extra call (eth_blockNumber) in the Connect
method to check that the client can obtain at least the
block number from the connected web3 endpoint.

jordipainan committed:
fix web3 connection sync routine if nethermind:

Adapt the sync progress routine to work properly
when the connected web3 endpoint runs the Nethermind
codebase. The eth_syncing JSON RPC call returns 0x0
as block number.

Add an extra call (eth_blockNumber) in the Connect
method to check that the client can obtain at least the
block number from the connected web3 endpoint.

p4u commented on ethereum/handler/ethereumHandler.go:
```diff
@@ -134,8 +142,22 @@ func (eh *EthereumHandler) SyncInfo(ctx context.Context) (*EthSyncInfo, error) {
+		defer cancel()
+		blkNum, err := eh.EthereumClient.BlockNumber(ctx)
+		if err != nil {
+			log.Debugf("cannot connect: %s", err)
```
if the error is returned it should not need to be printed

p4u commented on ethereum/handler/ethereumHandler.go:
```diff
@@ -134,8 +142,22 @@ func (eh *EthereumHandler) SyncInfo(ctx context.Context) (*EthSyncInfo, error) {
+		// do an extra check getting the latest block number via eth_blockNumber
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+		defer cancel()
+		blkNum, err := eh.EthereumClient.BlockNumber(ctx)
```
use info.Height directly

p4u commented on ethereum/handler/ethereumHandler.go:
```diff
@@ -134,8 +142,22 @@ func (eh *EthereumHandler) SyncInfo(ctx context.Context) (*EthSyncInfo, error) {
 		return nil, err
 	}
 	if sp != nil {
```
lets reverse this
```
if sp == nil {
	header, err := eh.EthereumClient.HeaderByNumber(ctx, nil)
		if err != nil {
			return nil, err
		}
		info.Height = uint64(header.Number.Int64())
		info.MaxHeight = info.Height
		info.Synced = info.Height > 0
		return info,nil
		}
```

p4u commented on ethereum/handler/ethereumHandler.go:
```diff
@@ -133,17 +141,28 @@ func (eh *EthereumHandler) SyncInfo(ctx context.Context) (*EthSyncInfo, error) {
 		info.Height = uint64(header.Number.Int64())
 		info.MaxHeight = info.Height
 		info.Synced = info.Height > 0
+	} else {
```
The objective of the if reverse was to remove the "else" and add a return inside the if

jordipainan replied:
done

p4u committed:
rpcapi: register handler for getprocessRollingCensusSize

Signed-off-by: p4u <pau@dabax.net>

ed255 committed:
Update genesis circuits config

p4u committed:
testsuit: fix race condition

Due the new httpRouter implementation, the http server is available
quite before the rest of the API methods are registered into the router.

The testsuit start_test.sh script only checks if the http server is
listening or not (before starting the test). If the server is fast
enough generating the test keys (first step) it tries do execute the
addCensus API method before the gateway has it available.

This commit introduces a workaound (sleep of 10 seconds after http
server found ready) that should work for now.

Signed-off-by: p4u <pau@dabax.net>
